ethereum/rpc/backend/backend.go

```diff
@@ -294,8 +295,23 @@ func (e *EVMBackend) GetTransactionLogs(txHash common.Hash) ([]*ethtypes.Log, er
 
 // PendingTransactions returns the transactions that are in the transaction pool
 // and have a from address that is one of the accounts this node manages.
-func (e *EVMBackend) PendingTransactions() ([]*types.RPCTransaction, error) {
-	return []*types.RPCTransaction{}, nil
+func (e *EVMBackend) PendingTransactions() ([]*sdk.Tx, error) {
+	res, err := e.clientCtx.Client.UnconfirmedTxs(e.ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	result := make([]*sdk.Tx, 0, len(res.Txs))
+	for _, txBz := range res.Txs {
+		tx, err := e.clientCtx.TxConfig.TxDecoder()(txBz)
+		if err != nil {
+			result = append(result, nil)
+			continue
+		}
```

Comment:
I really think we should log and return the error here, no transaction should fall under this case. If it does then there is something wrong

Reply:
agree, done